etcd.py

```diff
@@ -97,42 +94,42 @@ def set_info_from_etcd(self, info):
 
         addr = self.get_addr_from_urls(info['peerURLs'])
         # when you add new member it doesn't have name, but we can match it by peer_addr
-        if self.addr and (not addr or self.addr != addr):
+        if self.dns and (not addr or self.dns != addr):
             return
 
         self.id = info['id']
         self.name = info['name']
         self.client_urls = info['clientURLs']
         self.peer_urls = info['peerURLs']
-        self.addr = addr
+        self.dns = addr
 
     @staticmethod
     def generate_url(addr, port):
-        return 'http://{}:{}'.format(addr, port)
+        return 'https://{}:{}'.format(addr, port)
 
     def get_client_url(self, endpoint=''):
-        url = self.generate_url(self.addr, self.client_port)
+        url = self.generate_url(self.dns, self.client_port)
         if endpoint:
             url += self.API_VERSION + endpoint
         return url
 
     @property
     def peer_addr(self):
-        return '{}:{}'.format(self.addr, self.peer_port)
+        return '{}:{}'.format(self.dns, self.peer_port)
 
     @property
     def peer_url(self):
-        return self.generate_url(self.addr, self.peer_port)
+        return self.generate_url(self.dns, self.peer_port)
 
     def api_get(self, endpoint):
         url = self.get_client_url(endpoint)
-        response = requests.get(url, timeout=self.API_TIMEOUT)
+        response = requests.get(url, timeout=self.API_TIMEOUT, verify=False)
```

Comment:
The service runs on the same machine like the etcd itself. Is it not possible, to use this ca?